### ISSUE TITLE
Accessibility issues with canvas elements

### ISSUE THREAD

**Issue by jvine** [ProjectMirador/mirador#2935]
AMP is reporting a couple of violations related to canvases.

### 1. Missing labels
> **Violation**: Ensure embedded elements and canvas elements provide a meaningful text equivalent. 
> **Note**: This canvas element does not have a mechanism that allows an accessible name value to be calculated

The containing `<section>` has an `aria-label` (??) but the `<canvas>` element itself does not. Maybe more efficient to remove the label from the `<section>` and use `aria-labelledby` on the `<canvas>` element to refer to the text shown in the visible caption `<div class="mirador-osd-info">`.

Complication:  there seem to be two canvases per `<section>`. We don't want to label both like this. It's not clear to me if both are perceived by AT as opposed to being picked up by automated testing. I don't know what the equivalent of a null alt might be for a canvas label. 

### 2. Height/width specification
> **Violation**: Ensure containing elements allow text resize without loss of functionality.
> **Note**: The width attribute should not be used on this element
> **Note**: The height attribute should not be used on this element

Instances look like this:
`<canvas width="1480" height="2274" style="background: none transparent; border: none; margin: 0px; padding: 0px; position: absolute; width: 100%; height: 100%;"></canvas>`
`<canvas width="740" height="1137"></canvas>`

I don't understand this one. `<canvas>` elements aren't meant to contain text in the first place. Not sure if this is just an automated testing artifact that can be ignored or if it has a real effect. Can anyone with more a11y chops confirm?

**Comment by brittnylapierre** [comment]
1. Fixed as of 2026-04-24

**Comment by brittnylapierre** [comment]
2. What the checker is complaining about is usually WCAG 1.4.4 (Resize Text) or sometimes 1.4.10 (Reflow). 

Automated tools sometimes can flag fixed width/height attributes on all elements because they can interfere with zoom/reflow behavior. For <canvas> width and height are needed to keep the rendering of the content sharp and not blurry. Using only CSS stretches the default \(300 \times 150\) surface, resulting in blurry, pixilated graphics.
